src/MongoDB.Driver.Legacy/Builders/IndexKeysBuilder.cs

```diff
@@ -66,6 +66,7 @@ public static IndexKeysBuilder GeoSpatial(string name)
         /// </summary>
         /// <param name="name">The key name.</param>
         /// <returns>The builder (so method calls can be chained).</returns>
+        [Obsolete("The geoHaystack was deprecated in server version 4.4.")]
```

Comment:
Replace all Obsolete messages with this one that reads better:

    [Obsolete("GeoHaystack indexes were deprecated in server version 4.4.")]

Reply:
TODO: Obsolete all occurrences of `BucketSize` with the same message.

BucketSize is only used by GeHaystack indexes.

Reply:
done